Toolchain: Move nix script from Documentation into Toolchain (#12940)
SerenityOS/serenity: Documentation/BuildInstructionsOther.md

## NixOS

> You can use this `nix-shell` script [Toolchain/serenity.nix](../Toolchain/serenity.nix) to setup the correct environment with `nix-shell Toolchain/serenity.nix`.

Reviewer: ````suggestion
You can use the `nix-shell` script [`Toolchain/serenity.nix`](../Toolchain/serenity.nix) to set up the environment:

```console
nix-shell Toolchain/serenity.nix
```
````
